.github/workflows/ci.yml

```diff
@@ -0,0 +1,226 @@
+      - name: "Setup environment"
+        uses: "networktocode/gh-action-setup-poetry-environment@v1"
+      - name: "Linting: flake8"
+        run: "poetry run invoke flake8"
+  hadolint:
+    runs-on: "ubuntu-20.04"
+    env:
+      INVOKE_NAUTOBOT_LOCAL: "True"
+    steps:
+      - name: "Check out repository code"
+        uses: "actions/checkout@v2"
+      - name: "Setup environment"
+        uses: "networktocode/gh-action-setup-poetry-environment@v1"
+      - name: "Install hadolint"
+        run: "curl -Lo /usr/local/bin/hadolint https://github.com/hadolint/hadolint/releases/download/v2.0.0/hadolint-Linux-x86_64 && chmod +x /usr/local/bin/hadolint"
+      - name: "Linting: hadolint"
+        run: "poetry run invoke hadolint"
+  check-migrations:
+    runs-on: "ubuntu-20.04"
+    env:
+      INVOKE_NAUTOBOT_LOCAL: "True"
+      NAUTOBOT_DB_USER: "nautobot"
+      NAUTOBOT_DB_PASSWORD: "decinablesprewad"
+      NAUTOBOT_DB_ENGINE: "django.db.backends.postgresql"
+    services:
+      postgres:
+        image: "postgres:13"
+        env:
+          POSTGRES_USER: "nautobot"
+          POSTGRES_PASSWORD: "decinablesprewad"
+          POSTGRES_DB: "nautobot"
+        ports:
+          - "5432:5432"
+    steps:
+      - name: "Check out repository code"
+        uses: "actions/checkout@v2"
+      - name: "Setup environment"
+        uses: "networktocode/gh-action-setup-poetry-environment@v1"
+      - name: "Run check migrations"
+        run: "poetry run invoke check-migrations"
+    needs:
+      - "black"
+      - "flake8"
+      - "hadolint"
+  tests-postgres:
+    runs-on: "ubuntu-20.04"
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version: [ "3.6", "3.7", "3.8", "3.9" ]
+    env:
+      INVOKE_NAUTOBOT_LOCAL: "True"
+      NAUTOBOT_DB_USER: "nautobot"
+      NAUTOBOT_DB_PASSWORD: "decinablesprewad"
+      NAUTOBOT_DB_ENGINE: "django.db.backends.postgresql"
+    services:
+      postgres:
+        image: "postgres:13"
+        env:
+          POSTGRES_USER: "nautobot"
+          POSTGRES_PASSWORD: "decinablesprewad"
+          POSTGRES_DB: "nautobot"
+        ports:
+          - "5432:5432"
+      redis:
+        image: "redis:6-alpine"
+        ports:
+          - "6379:6379"
+    steps:
+      - name: "Check out repository code"
+        uses: "actions/checkout@v2"
+      - name: "Set up Python"
+        uses: "actions/setup-python@v2"
+        id: "python-setup"
+        with:
+          python-version: "${{ matrix.python-version }}"
+      - name: "Install Poetry"
+        run: "pip install poetry"
+      - name: "Load Environment From Cache"
+        uses: "actions/cache@v2"
+        id: "cached-poetry"
+        with:
+          path: "~/.cache/pypoetry/virtualenvs/"
+          key: "${{ runner.os }}-poetry-${{ steps.python-setup.outputs.python-version }}-${{ hashFiles('./poetry.lock') }}"
+      - name: "Install Dependencies"
+        run: "poetry install --no-interaction --no-root"
```

Comment:
Can we cache the dependencies?

Reply:
Why can't/shouldn't we use networktocode/gh-action-setup-poetry-environment for these actions as well?

Reply:
@nniehoff, actually, we do use cached dependencies. Because we are using the matrix (python versions 3.6, 3.7, 3.8, 3.9), it creates four caches.

The following line, "Install Dependencies" is just a precaution for the cases when the cache is not found. If the cache is found, then the result of the command is "No dependencies to install or update".

@glennmatthews, we can, but first, I need to update the action to accept the python version as the input parameter. I am currently using the hardcoded python version in the GH action.

But I agree, it makes sense.

Reply:
@glennmatthews I have updated the GH action. It now accepts the python version as well. So I am using the same approach as for other steps.

Reply:
Can we also cache the poetry install?  I see the poetry install in every stage

Reply:
This is provided in the latest version...